src-tauri/src/manager.rs

```diff
@@ -109,6 +124,18 @@ impl ManagerState {
             }
         }
     }
+    pub fn handle_system_click(&mut self,name: &'static str) {
+        if self.is_watcher_running(name){
+            self.stop_watcher(name);
+        } else {
+            let mut state = WATCHER_STATE.lock().unwrap();
+            state.push(name);
+            let (lock, condvar) = &*WATCHER_CONDVAR;
+            let mut l = lock.lock().unwrap();
+            *l = true;
+            condvar.notify_all();
+        }
+    }
```

Comment:
Only the name of watcher is passed when the system_tray is clicked. We check if the passed ```name``` is running, if it is we stop it. Otherwise we start it by pushing it to WATCHER_STATE. And alert the waiting ```wrapped_start_watcher``` function via the condition variable.

Reply:
Leave comments like this in the code!

Reply:
Removed this too, now simply calls `self.start_watcher()`